src/bentoml/_internal/container/kaniko.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+import os
+import shutil
+import typing as t
+import logging
+from typing import TYPE_CHECKING
+
+from .base import Arguments
+from ..utils import resolve_user_filepath
+
+if TYPE_CHECKING:
+    from .base import ArgType
+    from ..types import PathType
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "health",
+    "construct_build_args",
+    "BUILDKIT_SUPPORT",
+    "BUILD_CMD",
+    "find_binary",
+]
+
+BUILDKIT_SUPPORT = False
+
+BUILD_CMD = []
+
+
+def find_binary() -> str | None:
+    # We will check if /kaniko/executor exists
+    if os.path.exists("/kaniko/executor"):
+        return "/kaniko/executor"
+    return shutil.which("executor")
+
```

Comment:
this will check for kaniko in the root filesystem right but Kaniko is designed to be run as a container?

Reply:
I think I make an assumption here that kaniko will be ran inside a container.

L35 are mostly act as a fallback when I was debugging and building executor from a container and move it to a different path